### PR TITLE
render/egl: add support for EGL_EXT_device_drm_render_node

### PR DESCRIPTION
**Pull request by emersion**
This EGL extension has been added in [1]. The upsides are:

- We directly get a render node, instead of having to convert the
  primary node name to a render node name.
- If EGL_DRM_RENDER_NODE_FILE_EXT returns NULL, that means there is
  no render node being used by the driver.

[1]: https://github.com/KhronosGroup/EGL-Registry/pull/127

Mesa patch: https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/11797
libglvnd patch: https://gitlab.freedesktop.org/glvnd/libglvnd/-/merge_requests/245

### CODE REVIEW

**Comment by emersion** [comment]
The Mesa and libglvnd patches have been merged.

File: render/egl.c
Comment (bl4ckb0ne) [inline render/egl.c:723]: Why the preprocessor? Isn't checking `egl->exts.EXT_device_drm_render_node` enough?

Comment (emersion) [reply render/egl.c:723]: `EGL_EXT_device_drm_render_node` has been merged recently, so won't be widely available in the system headers for a while. Need to wait for new libglvnd and Mesa releases.

Comment (bl4ckb0ne) [reply render/egl.c:723]: So we're waiting for a release before merging?

Comment (emersion) [reply render/egl.c:723]: No need, the `#ifdef` ensures this code is disabled for older Mesa/libglvnd, and is enabled for recent ones.

**Comment by emersion** [comment]
We may want to merge https://github.com/swaywm/wlroots/pull/2671 first.

File: render/egl.c
Comment (bl4ckb0ne) [inline render/egl.c:718]: Wont this check fail if the user doesnt have the ext?

Comment (emersion) [reply render/egl.c:718]: If the user doesn't have `EXT_device_drm_render_node`, but has `EXT_device_drm`, then the `if` is not taken. The `if` is only taken if the user is missing both exts.

Comment (bl4ckb0ne) [reply render/egl.c:718]: I should learn to read throughly the conditions for next time, sorry

**Review by bl4ckb0ne — APPROVED** [review]

**Comment by bl4ckb0ne** [comment]
LGTM, sorry for the dumb comment

File: render/egl.c
Comment (emersion) [reply render/egl.c:718]: No worries!

**Pull request merged**
